### DISCUSSION TITLE
Async call in update function

### DISCUSSION THREAD

**Discussion by fabuloso** [ivanceras/sauron#58] [Q&A] [answered]
Hello everyone, 
I'm trying to do a little application using Sauron, but I've encountered a problem.
I need to do an async http call by reqwest in the update function of the Application trait.
Unfortunately Application does not have `async_trait`, so I'm blocked.
Do you have any suggest?

**Comment by ivanceras** [comment, marked as answer]
Hi,
There are 2 ways to do `http` call in sauron one is using the provide `Http` module as demonstrated in this [example](https://github.com/ivanceras/sauron/blob/790359d55b73d60932d4b9089d62714364ab7806/examples/fetch-data/src/lib.rs).

The other way is suited for your use-case which make use of `reqwest` and `wasm-bindgen-futures`.
Take a look at this [hackernews-clone](https://github.com/ivanceras/hackernews-sauron/blob/28ba0549b291fa174c4279631e54ce5a571ff60e/client/src/app.rs#L306). This essentially spawns an `aync` function then wrap it in a `Cmd`

**Reply by fabuloso** [reply]
thanks a lot ❤️
